gcc/brig/brigfrontend/brig-to-generic.cc

```diff
@@ -555,7 +599,11 @@ build_stmt (enum tree_code code, ...)
    than the created reg var type in order to select correct instruction type
    later on.  This function creates the necessary reinterpret type cast from
    a source variable to the destination type.  In case no cast is needed to
-   the same type, SOURCE is returned directly.  */
+   the same type, SOURCE is returned directly.
+
+   In case of mismatched type sizes, casting:
```

Comment:
Ah OK, you moved to logic here. Then the function name is misleading as it sounds like no type-specific treatment is done, basically cast via pointer type casting. However, it's not a proper C cast either as we don't want float to int conversions to happen numerically. build_conversion() or similar?

Reply:
build_resize_convert_view()? Its what VIEW_CONVERT_EXPR does plus the added semantics here.